### PR TITLE
[WIP] Add cache unit tests

### PR DESCRIPTION
**Pull request by foursixnine**
This is a first approach of unit tests for the cache, to improve
coverage and test basic functionality.

When the error code is not present in $tx->res perl
complains, particularily in "Connection Refused" cases.

### CODE REVIEW

**Comment by foursixnine** [comment]
Let's see how it all breaks :)

**Comment by foursixnine** [comment]
Go home @codecov[bot]  you're drunk!

File: t/25-cache.t [hunk 1]
```diff
@@ -0,0 +1,177 @@
+#! /usr/bin/perl
+
+# Copyright (c) 2015 SUSE LINUX GmbH, Nuernberg, Germany.
```
Comment (coolo) [inline t/25-cache.t]: interesting choice of year

File: t/25-cache.t [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+
+use DBI;
+
+use Data::Dump qw(pp dd);
+use File::Path qw(remove_tree make_path);
+use File::Spec::Functions 'catfile';
+use Digest::MD5 qw(md5);
+
+use Cwd qw(abs_path getcwd);
+use File::Spec::Functions 'catdir';
```
Comment (coolo) [inline t/25-cache.t]: better join the functions from ::Functions

Comment (kraih) [inline t/25-cache.t]: Or just use `Mojo::File`. 😉

File: lib/OpenQA/Worker/Cache.pm [hunk 3]
```diff
@@ -350,14 +350,20 @@ sub check_limits {
...
         $result = $dbh->selectrow_hashref($sql);
-
-        foreach my $asset ($result) {
-            if (purge_asset($asset->{filename})) {
-                $cache_real_size -= $asset->{size};
-                log_debug "Reclaiming " . $asset->{size} . " from $cache_real_size to make space for $limit";
-            }    # purge asset will die anyway in case of failure.
-            last if ($cache_real_size < $limit);
+        if ($result) {
+            foreach my $asset ($result) {
```
Comment (kraih) [inline lib/OpenQA/Worker/Cache.pm]: This loop appears to do nothing, what am i missing? Is aliasing `$result` to `$asset` somehow significant here?

**Comment by foursixnine** [comment]
I just noticed i really messed up somehow this :) Will reopen

**Pull request closed without merging**